packages/data/src/components/with-select/test/index.js

```diff
@@ -440,173 +399,147 @@ describe( 'withSelect', () => {
 		} );
 
 		const mapSelectToProps = jest.fn();
-
-		const OriginalComponent = jest.fn().mockImplementation( () => <div /> );
+		const OriginalComponent = jest.fn( () => <div /> );
 
 		const DataBoundComponent = compose( [
 			withSelect( mapSelectToProps ),
 		] )( OriginalComponent );
 
-		act( () => {
-			TestRenderer.create(
-				<RegistryProvider value={ registry }>
-					<DataBoundComponent />
-				</RegistryProvider>
-			);
-		} );
+		render(
+			<RegistryProvider value={ registry }>
+				<DataBoundComponent />
+			</RegistryProvider>
+		);
 
 		// 2 times:
 		// - 1 on initial render
 		// - 1 on effect before subscription set.
 		expect( mapSelectToProps ).toHaveBeenCalledTimes( 2 );
 		expect( OriginalComponent ).toHaveBeenCalledTimes( 1 );
 
-		store.dispatch( { type: 'dummy' } );
+		await act( async () => store.dispatch( { type: 'dummy' } ) );
 
 		expect( mapSelectToProps ).toHaveBeenCalledTimes( 2 );
 		expect( OriginalComponent ).toHaveBeenCalledTimes( 1 );
 	} );
 
 	it( 'omits props which are not returned on subsequent mappings', () => {
+		const registry = createRegistry();
 		registry.registerStore( 'demo', {
 			reducer: ( state = 'OK' ) => state,
 			selectors: {
 				getValue: ( state ) => state,
 			},
 		} );
 
-		const mapSelectToProps = jest
-			.fn()
-			.mockImplementation( ( _select, ownProps ) => {
-				return {
-					[ ownProps.propName ]: _select( 'demo' ).getValue(),
-				};
-			} );
+		const mapSelectToProps = jest.fn( ( _select, ownProps ) => {
+			return {
+				[ ownProps.propName ]: _select( 'demo' ).getValue(),
+			};
+		} );
 
-		const OriginalComponent = jest
-			.fn()
-			.mockImplementation( ( props ) => (
-				<div>{ JSON.stringify( props ) }</div>
-			) );
+		const OriginalComponent = jest.fn( ( props ) => (
+			<div>{ JSON.stringify( props ) }</div>
+		) );
 
 		const DataBoundComponent =
 			withSelect( mapSelectToProps )( OriginalComponent );
 
-		let testRenderer;
-		act( () => {
-			testRenderer = TestRenderer.create(
-				<RegistryProvider value={ registry }>
-					<DataBoundComponent propName="foo" />
-				</RegistryProvider>
-			);
-		} );
-		const testInstance = testRenderer.root;
+		const { container, rerender } = render(
+			<RegistryProvider value={ registry }>
+				<DataBoundComponent propName="foo" />
+			</RegistryProvider>
+		);
 
 		// 2 times:
 		// - 1 on initial render
 		// - 1 on effect before subscription set.
 		expect( mapSelectToProps ).toHaveBeenCalledTimes( 2 );
 		expect( OriginalComponent ).toHaveBeenCalledTimes( 1 );
 
-		expect(
-			JSON.parse( testInstance.findByType( 'div' ).props.children )
-		).toEqual( {
-			foo: 'OK',
-			propName: 'foo',
-		} );
+		expect( container ).toHaveTextContent(
+			JSON.stringify( {
```

Comment:
That feels a bit fragile and actually close to what a snapshot test would do. I wonder if a snapshot test is a cleaner way to achieve the same. WDYT?

Reply:
The `useSelect` has a similar assertion, so I tried not to change this one too much.

Reply:
Sounds fine to me. It's not a huge deal since we're instrumenting this component ourselves, and not really querying against a production component.

Reply:
> The `useSelect` has a similar assertion, so I tried not to change this one too much.

It was me who wrote these `useSelect` tests a few months ago 🙂 and yes, I had to think for a while how to test for an arbitrary JSON value. I ended up serializing the values with `JSON.stringify`, as DOM attributes can only be strings and nothing else, except maybe numbers. There's no way to store an arbitrary JS value in the DOM.

Only I used a `data-d` attribute, and checked for `screen.getByRole( 'status' ).dataset.d`.